core/src/main/java/org/infinispan/persistence/async/AsyncNonBlockingStore.java

```diff
@@ -473,72 +459,70 @@ public void run() {
    public CompletionStage<Void> batch(int publisherCount, Publisher<SegmentedPublisher<Object>> removePublisher,
          Publisher<SegmentedPublisher<MarshallableEntry<K, V>>> writePublisher) {
       assertNotStopped();
-      Flowable.fromPublisher(removePublisher)
-            .subscribe(sp ->
-               Flowable.fromPublisher(sp)
-                     .subscribe(key -> submissionFlowable.onNext(new RemoveModification(sp.getSegment(), key)))
-            );
-      Flowable.fromPublisher(writePublisher)
-            .subscribe(sp ->
-                  Flowable.fromPublisher(sp)
-                        .subscribe(me -> submissionFlowable.onNext(new PutModification(sp.getSegment(), me)))
-            );
-      submitBatchIfNecessary();
-      return asyncOrThrottledStage();
-   }
-
-   @Override
-   public CompletionStage<Void> write(int segment, MarshallableEntry<? extends K, ? extends V> entry) {
-      assertNotStopped();
-      submissionFlowable.onNext(new PutModification(segment, entry));
-      submitBatchIfNecessary();
-      return asyncOrThrottledStage();
-   }
-
-   @Override
-   public CompletionStage<Boolean> delete(int segment, Object key) {
-      assertNotStopped();
-      submissionFlowable.onNext(new RemoveModification(segment, key));
-      submitBatchIfNecessary();
-      return asyncOrThrottledStage()
-            // We always assume it was removed with async
-            .thenCompose(ignore -> CompletableFutures.completedTrue());
-   }
-
-   private void submitBatchIfNecessary() {
+      // Note that all of these use the lastElement method to detect completion. This is okay only because the
```

Comment:
You removed the `lastElement()` call, so I think you can also remove this comment.

Reply:
Sure.